### PR TITLE
Issue #1328 Print 'oc cluster up' options

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

**Comment by gbraad** [comment]
add to whitelist

**Comment by thomasmckay** [comment]
@gbraad - This seemed least intrusive while still being useful. Shall I keep my changes and add glog to this file or to cmd/start.go?

**Comment by gbraad** [comment]
Adding `glog` sounds good. After this I will try a test run and see if additional change might be needed.

Note: I do not want to really show TOO much information to the user (clutters). Maybe the values can be printed only when the verbosity level is increased?

**Comment by praveenkumar** [comment]
@thomasmckay Do you need any help with this regard,  PR is not updated for quite some time now.

**Comment by thomasmckay** [comment]
My suggestion is to add this line before runner.Run:
```
log.Debug(fmt.Sprintf("-- Running `oc` with: '%s'", strings.Join(cmdArgs, " ")))
```
I could not get glog.V(5).Info() to work w/ args of --v 5 --show-libmachine-logs. 

Thoughts?

File: pkg/minishift/clusterup/clusterup.go [hunk 1]
```diff
@@ -99,6 +100,7 @@ func ClusterUp(config *ClusterUpConfig, clusterUpParams map[string]string, runne
 		}
 	}
 
+        log.Debug(fmt.Sprintf("-- Running `oc` with: '%s'", strings.Join(cmdArgs, " ")))
```
Comment (praveenkumar) [inline pkg/minishift/clusterup/clusterup.go]: @thomasmckay extra space begin of this log.Debug.

@LalatenduMohanty @gbraad I can see the command args for oc cluster up when using below which I think what we anticipating here.

```
$ minishift start --show-libmachine-logs --v=5
[...]
About to run SSH command:
sudo mkdir -p /var/lib/minishift/openshift.local.config /var/lib/minishift/hostdata /var/lib/minishift/openshift.local.volumes /var/lib/minishift/openshift.local.pv
SSH cmd err, output: <nil>: 
-- OpenShift cluster will be configured with ...
   Version: v3.6.0
-- Checking `oc` support for startup flags ... 
   host-volumes-dir ... OK
   routing-suffix ... OK
   host-config-dir ... OK
   host-data-dir ... OK
   host-pv-dir ... OK
-- Running `oc` with: 'cluster up --use-existing-config --host-volumes-dir /var/lib/minishift/openshift.local.volumes --routing-suffix 192.168.42.79.nip.io --host-config-dir /var/lib/minishift/openshift.local.config --host-data-dir /var/lib/minishift/hostdata --host-pv-dir /var/lib/minishift/openshift.local.pv'
[...]
```

Comment (LalatenduMohanty) [reply pkg/minishift/clusterup/clusterup.go]: @praveenkumar @thomasmckay Tried the PR and it looks good to me.

**Comment by LalatenduMohanty** [comment]
We need to fix the go fmt of the PR (just run `make fmt` and it will fix it for you) and rebase it with current master.

**Comment by LalatenduMohanty** [comment]
@thomasmckay if you use glog as below it works fine. I recently found out. Need to check why `glog.V(2).Info` does not work for me. 

```
if glog.V(2) {
    anything
}

```

**Comment by LalatenduMohanty** [comment]
The advantage of using glog is that we do not have to depend on --show-libmachine-logs for this log i.e. if user uses --v 2 or anything >2 user would be able to see the log.

File: pkg/minishift/clusterup/clusterup.go [hunk 2]
```diff
@@ -99,6 +100,9 @@ func ClusterUp(config *ClusterUpConfig, clusterUpParams map[string]string, runne
 		}
 	}
 
+	if glog.V(5) {
+		fmt.Println("-- Running `oc` with: '%s'", strings.Join(cmdArgs, " "))
```
Comment (LalatenduMohanty) [inline pkg/minishift/clusterup/clusterup.go]: @thomasmckay fmt.Println can not print formatted strings. So you need to use fmt.Sprintf i.e. `fmt.Println(fmt.Sprintf("-- Running `oc` with: '%s'", strings.Join(cmdArgs, " ")))`

Comment (LalatenduMohanty) [reply pkg/minishift/clusterup/clusterup.go]: Also I think `fmt.Println(fmt.Sprintf("-- Running oc '%s'", strings.Join(cmdArgs, " ")))` would be better.

Comment (praveenkumar) [reply pkg/minishift/clusterup/clusterup.go]: > Also I think fmt.Println(fmt.Sprintf("-- Running oc '%s'", strings.Join(cmdArgs, " "))) would be better

I would say use `fmt.Printf("-- Running `oc` with: '%s'\n", strings.Join(cmdArgs, " "))` instead calling 2 different functions.

Comment (gbraad) [reply pkg/minishift/clusterup/clusterup.go]: `fmt.Printf("%s\n",sth)` is the preferred approach

Comment (budhrg) [reply pkg/minishift/clusterup/clusterup.go]: > I would say use fmt.Printf("-- Runningocwith: '%s'\n", strings.Join(cmdArgs, " ")) instead calling 2 different functions.

:+1:  This is fine @thomasmckay

Comment (thomasmckay) [reply pkg/minishift/clusterup/clusterup.go]: switched to fmt.Printf()

**Comment by gbraad** [comment]
Ready to merge, but seems CentOS-CI is not progressing...

**Pull request merged**